File: scala/p7.scala
// Solution for Project Euler problem 7
// http://projecteuler.net/index.php?section=problems&id=7

import scala.Math._

def prime(n: Int): Boolean =
  if ( (2 to pow(n, .5).toInt).reverse.filter(x => n % x == 0).length == 0 )
    true
  else
    false

def nextprime(n: Int): Int = n match {
  case x if prime(x) => x
  case x => nextprime(x + 2)
}

lazy val primes: Stream[Int] = Stream.cons(3, primes.map(x => nextprime(x + 2)))

// Seeded with the 2nd prime
println(primes.apply(10001 - 2))

File: python/p7.py
# Solution for Project Euler problem 7
# http://projecteuler.net/index.php?section=problems&id=7

# i = 3
# q = [2]
# while len(q) < 10001:
#     prime = True
#     for j in q:
#         if i % j == 0:
#             prime = False
#             break
#     if prime:
#         q.append(i)
#     i += 2

# print q[-1]

def prime(n):
    for i in xrange(int(n ** .5), 1, -1):
        if n % i == 0:
            return False
    return True

count = 2
n = 3
while count < 10001:
    n += 2
    if prime(n):
        count += 1
print n
